app/email-incoming/index.ts

```diff
@@ -133,10 +118,47 @@ export async function handler(event: SNSEvent) {
 }
 
 /**
- * Returns an entry from the legacy_users table, or undefined if none exists
+ * Returns a UserData object constructed from cognito if the
+ * user is in the Submitters group
+ * @param userEmail
+ */
+async function getCognitoUserData(
+  userEmail: string
+): Promise<UserData | undefined> {
+  const data = await cognito.send(
+    new ListUsersInGroupCommand({
+      GroupName: group,
+      UserPoolId: process.env.COGNITO_USER_POOL_ID,
+    })
+  )
+  const userTypeData = data.Users?.find(
+    (user) => extractAttributeRequired(user, 'email') == userEmail
+  )
+  return userTypeData
+    ? {
+        sub: extractAttributeRequired(userTypeData, 'sub'),
+        email: extractAttributeRequired(userTypeData, 'email'),
+        name: extractAttribute(userTypeData, 'name'),
+        affiliation: extractAttribute(userTypeData, 'custom:affiliation'),
+      }
+    : undefined
```

Comment:
Two slightly shorter options:

```suggestion
  if (userTypeData)
    return {
        sub: extractAttributeRequired(userTypeData, 'sub'),
        email: extractAttributeRequired(userTypeData, 'email'),
        name: extractAttribute(userTypeData, 'name'),
        affiliation: extractAttribute(userTypeData, 'custom:affiliation'),
      }
```

or

```suggestion
  return userTypeData && {
      sub: extractAttributeRequired(userTypeData, 'sub'),
      email: extractAttributeRequired(userTypeData, 'email'),
      name: extractAttribute(userTypeData, 'name'),
      affiliation: extractAttribute(userTypeData, 'custom:affiliation'),
    }
```